src/transcripts.rs

```diff
@@ -36,26 +36,30 @@ use crate::{
 /// When randomness is needed, just use the `TranscriptRng`.
 /// The prover uses this whenever it needs a random nonce.
 /// The batch verifier uses this to generate weights.
-pub(crate) struct RangeProofTranscript<P>
+pub(crate) struct RangeProofTranscript<'a, P, R>
 where
     P: Compressable + Precomputable,
     P::Compressed: FixedBytesRepr + IsIdentity,
+    R: CryptoRngCore,
 {
     transcript: Transcript,
     bytes: Option<Zeroizing<Vec<u8>>>,
+    rng: TranscriptRng,
```

Comment:
I think the previous naming is better suited to the intent. It is also still in line with the comment `/// When randomness is needed, just use the `TranscriptRng`.`
```suggestion
    transcript_rng: TranscriptRng,
```

Reply:
Updated.